src/coreclr/src/inc/corinfo.h

```diff
@@ -423,12 +423,10 @@ enum CorInfoHelpFunc
     // the right helper to use
 
     CORINFO_HELP_ISINSTANCEOFINTERFACE, // Optimized helper for interfaces
-    CORINFO_HELP_ISINSTANCEOFARRAY,  // Optimized helper for arrays
```

Comment:
I would keep these for now, even if it means that the implementation is same.

Reply:
I can get them back.

---
In reply to: [368260447](https://github.com/dotnet/runtime/pull/1068#discussion_r368260447) [](ancestors = 368260447)

Reply:
I just could not see what could be done differently in a helper if we knew we care casting to an array. 
For identity cast check we could just compare MTs, which we do for any cast...

---
In reply to: [368261248](https://github.com/dotnet/runtime/pull/1068#discussion_r368261248) [](ancestors = 368261248,368260447)

Reply:
I am guessing JIT could inline identity cast check similarly to class cast. Not sure if those are common with arrays but  with classes inlining identity cast helps a lot.

---
In reply to: [368261350](https://github.com/dotnet/runtime/pull/1068#discussion_r368261350) [](ancestors = 368261350,368261248,368260447)